ledger/participant-integration-api/src/main/scala/platform/store/dao/events/ContractsReader.scala

```diff
@@ -44,49 +44,66 @@ private[dao] sealed class ContractsReader(
       ),
     )
 
+  override def lookupContractStates(contractIds: Seq[ContractId], before: Offset)(implicit
+      loggingContext: LoggingContext
+  ): Future[Map[ContractId, ContractState]] = {
+    dispatcher
+      .executeSql(metrics.daml.index.db.lookupActiveContractDbMetrics)(
+        storageBackend.contractStates(contractIds, before)
+      )
+      .map(_.map { case (k, v) => (k, rawToContractState(k)(v)) })
```

Comment:
Wrong data in one of the rows i.e. RawContractState that is neither Create nor Archive will blow the future and thus the whole batch. I think it should be ok.